client/src/containers/CampaignAdmin/tests/CampaignAdmin-test.js

```diff
@@ -1,105 +1,210 @@
+      },
+    },
+    breadcrumbsActions: {
+      setBreadcrumbs: jest.fn(),
+    },
+    securityId: 'secured',
+    onResize: jest.fn(),
+    FormBuilderLoaderComponent: ({ onSubmit }) => <div
+      data-testid="test-form-builder-loader"
+      onClick={() => onSubmit({}, 'action_save', () => Promise.resolve(mockResponse))}
+    />,
+    BreadcrumbComponent: () => <div data-testid="test-breadcrumb"/>,
+    campaignActions: {
+      setNewItem: () => null
+    },
+    match: {
+      params: {
+        id: null,
+        view: null
+      }
+    },
+    sectionConfig: {
+      form: {
+        campaignCreateForm: {
+          schemaUrl: '/my/schema'
         }
-      };
-      admin = ReactTestUtils.renderIntoDocument(<CampaignAdmin {...props} />);
-    });
-    it('detects errors in errors property', () => {
-      const result = admin.hasErrors({
-        errors: [
-          { value: 'error' },
-        ],
-      });
-      expect(result).toBe(true);
-    });
+      },
+      reactRoutePath: '/campaigns',
+    },
+    router: {
+      params: {
+        view: 'create'
+      },
+      navigate: () => null
+    },
+    ...obj
+  };
+}
 
-    it('detects no errors in errors property', () => {
-      const result = admin.hasErrors({
-        errors: [],
-      });
-      expect(result).toBe(false);
-    });
+test('CampaignAdmin navigates to the show view when there are no errors', async () => {
+  const navigate = jest.fn();
+  render(<CampaignAdmin {...makeProps({
+    router: {
+      ...makeProps().router,
+      navigate
+    }
+  })}
+  />);
+  const loader = await screen.findByTestId('test-form-builder-loader');
+  // There is is an onClick handler on the mock FormBuilderLoaderComponent that acts as a proxy for onSubmit()
+  // The click event is only wrapped in a setTimeout so that it's fired a tick later which allows the mockResponse to be
+  // written after the click. This mean that the test reads sequentially.
+  setTimeout(() => fireEvent.click(loader));
+  mockResponse = {
+    errors: [],
+    record: {
+      id: 1
+    }
+  };
+  await screen.findByTestId('test-breadcrumb');
+  expect(navigate).toBeCalledWith('/campaigns/set/1/show');
+});
 
-    it('detects errors in global messages property', () => {
-      const result = admin.hasErrors({
-        state: {
-          messages: [
-            { value: 'error' },
-          ],
-        },
-      });
-      expect(result).toBe(true);
-    });
+test('CampaignAdmin detects errors in errors property', async () => {
+  const navigate = jest.fn();
+  render(<CampaignAdmin {...makeProps({
+    router: {
+      ...makeProps().router,
+      navigate
+    }
+  })}
+  />);
+  const loader = await screen.findByTestId('test-form-builder-loader');
+  setTimeout(() => fireEvent.click(loader));
+  mockResponse = {
+    errors: [
+      { value: 'error' },
+    ],
+    record: {
+      id: 1
+    }
+  };
+  await screen.findByTestId('test-breadcrumb');
+  expect(navigate).not.toBeCalled();
```

Comment:
Could we check `errors in errors property`?

Reply:
Errors in is a mock response, it doesn't make any sense to check it